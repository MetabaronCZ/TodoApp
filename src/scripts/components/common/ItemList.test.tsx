import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { ItemList } from 'components/common/ItemList';
import { withMockedProviders } from 'test/component';

describe('components/common/ItemList', () => {
  it('should render correctly', () => {
    const { container } = render(
      withMockedProviders(<ItemList>CONTENT</ItemList>),
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelector('ul')).toBeInTheDocument();

    const items = container.querySelectorAll('li');
    expect(items.length).toEqual(1);
    expect(items[0].textContent).toEqual('CONTENT');
  });

  it('should render nothing when no children', () => {
    const { container } = render(withMockedProviders(<ItemList />));
    expect(container).toMatchSnapshot();
    expect(container.innerHTML).toEqual('');
  });

  it('should render children', () => {
    const { container } = render(
      withMockedProviders(
        <ItemList>
          <span>1</span>
          <span>2</span>
          <span>3</span>
        </ItemList>,
      ),
    );
    expect(container).toMatchSnapshot();

    const items = Array.from(container.querySelectorAll('li'));
    expect(items.length).toEqual(3);
    expect(items.map((item) => item.textContent)).toEqual(['1', '2', '3']);
  });

  it('should render loading state', () => {
    const tree = render(
      withMockedProviders(<ItemList loading>CONTENT</ItemList>),
    );
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByRole('list')).toHaveStyle({ opacity: 0.5 });
  });
});
