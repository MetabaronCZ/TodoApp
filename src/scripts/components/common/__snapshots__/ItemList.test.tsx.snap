// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/common/ItemList > should render children 1`] = `
.c0 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-bottom: 1px solid #c0c0c0;
}

.c1:last-child {
  border-bottom: none;
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <span>
        1
      </span>
    </li>
    <li
      class="c1"
    >
      <span>
        2
      </span>
    </li>
    <li
      class="c1"
    >
      <span>
        3
      </span>
    </li>
  </ul>
</div>
`;

exports[`components/common/ItemList > should render correctly 1`] = `
.c0 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-bottom: 1px solid #c0c0c0;
}

.c1:last-child {
  border-bottom: none;
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      CONTENT
    </li>
  </ul>
</div>
`;

exports[`components/common/ItemList > should render loading state 1`] = `
.c0 {
  opacity: 0.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-bottom: 1px solid #c0c0c0;
}

.c1:last-child {
  border-bottom: none;
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      CONTENT
    </li>
  </ul>
</div>
`;

exports[`components/common/ItemList > should render nothing when no children 1`] = `<div />`;
